modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/Inbox.java

```diff
@@ -216,7 +216,7 @@ private void push() throws IgniteCheckedException {
     /** */
     private void pushOrdered() throws IgniteCheckedException {
          PriorityQueue<Pair<Row, Buffer>> heap =
-            new PriorityQueue<>(buffers.size(), Map.Entry.comparingByKey(comp));
+            new PriorityQueue<>(Math.max(buffers.size(), 1), Map.Entry.comparingByKey(comp));
```

Comment:
i understand why you refactor here: Math.max, but why do we need additional allocation here ? 
if (buffers.isEmpty)
  PriorityQueue<Pair<Row, Buffer>> heap = null; and further F.isEmpty(heap) check would be match clear, isn`t it ?

Reply:
possibly, I'll rewrite this piece of code, honestly it was just a rude fast fix